__tests__/endpoint.test.js

```diff
@@ -0,0 +1,37 @@
+const db = require("../db/connection.js");
+const request = require('supertest');
+const app = require('../app.js');
+const seed = require('../db/seeds/seed.js');
+const testData = require('../db/data/test-data/');
+
+
+beforeEach(()=> seed(testData))
+afterAll(() => db.end());
+
+describe('/api/topics', ()=>{
+    test('responds with status code 200', ()=>{
+        return request(app).get('/api/topics')
+        .expect(200)
+    })
+    test("Should return an object with property treasures", () => {
+        return request(app).get("/api/topics").expect(200).then(result => {
+             expect(Array.isArray(result.body.topics)).toBe(true);
+        })
+    })
+    test("Should return an object with the property topics", () => {
+        return request(app).get("/api/topics").expect(200).then(result => {
+            expect(result).hasOwnProperty("topics");
+        })
+    })
+    test("Should return an nested object each with the property slug and description", () => {
+        return request(app).get("/api/topics").expect(200).then(result => {
+
+            result.body.topics.forEach(obj=>{
+                expect(obj).hasOwnProperty("slug")
```

Comment:
You could improve this test by changing the expect to be `expect(topic).toHaveProperty("slug", expect.any(String));` which would test both key and value at the same time, but this would be a refactor :)